### PR TITLE
refactor(Accordion): use CSS logical properties instead of physical values

### PR DESCRIPTION
**Pull request by Cata1989**
## Description

This PR aims to modernize the CSS by incorporating logical properties for improved adaptability and support for RTL content direction and vertical layout.

We have implemented CSS [animation for the summary HTML element](https://codepen.io/ZoranJambor/pen/jOorzPv?editors=1100) using block-size and [calc-size](https://github.com/WebKit/standards-positions/issues/348) (still in draft). If calc-size is supported:

CleanShot 2024-07-17 at 11 39 09@2x

the open/close animation will be handled via CSS instead of JavaScript. This enhances considerably the way the animation behaves on a vertical layout.

If calc-size() is not yet supported by the browser (feature flag is not enabled) animations will be handled via JavaScript, and, if there's a need for a vertical layout, we have added the possibility to disable all animation so the component works as expected, without shifting when opening or closing.

## Related Issue

N/A

## Screenshots (if applicable)

https://github.com/user-attachments/assets/baa3a706-4626-4a4c-afc1-ba11c770626e

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/Endava/BEEQ/blob/main/CONTRIBUTING.md) document.
- [x] I have read the [CODE OF CONDUCT](https://github.com/Endava/BEEQ/blob/main/CODE_OF_CONDUCT.md) document.
- [x] I have reviewed my own code.
- [x] I have tested the changes locally.
- [x] I have updated the documentation (if applicable).
- [x] I have added unit tests and e2e tests (if applicable).
- [x] I have requested reviews from relevant team members.

### CODE REVIEW

File: packages/beeq/src/components/accordion-group/scss/bq-accordion-group.scss [hunk 1]
```diff
@@ -6,4 +6,7 @@
 
 :host {
   @apply block;
+
+  direction: rtl;
```
Comment (endv-bogdanb) [inline packages/beeq/src/components/accordion-group/scss/bq-accordion-group.scss:10]:
```suggestion
```

File: packages/beeq/src/components/accordion-group/scss/bq-accordion-group.scss [hunk 2]
```diff
@@ -6,4 +6,7 @@
...
+  writing-mode: vertical-rl;
```
Comment (endv-bogdanb) [inline packages/beeq/src/components/accordion-group/scss/bq-accordion-group.scss:11]:
```suggestion
```

File: packages/beeq/src/components/accordion/scss/bq-accordion.scss [hunk 3]
```diff
@@ -17,13 +17,16 @@
     }
   }
 
+  direction: rtl;
```
Comment (endv-bogdanb) [inline packages/beeq/src/components/accordion/scss/bq-accordion.scss:20]:
```suggestion
```

File: packages/beeq/src/components/accordion/scss/bq-accordion.scss [hunk 4]
```diff
@@ -17,13 +17,16 @@
...
+  writing-mode: vertical-rl;
```
Comment (endv-bogdanb) [inline packages/beeq/src/components/accordion/scss/bq-accordion.scss:21]:
```suggestion
```

File: packages/beeq/src/components/accordion/bq-accordion.tsx [hunk 5]
```diff
@@ -232,7 +232,7 @@ export class BqAccordion {
             </slot>
           </div>
         </summary>
-        <div class="bq-accordion__body overflow-hidden" part="panel">
```
Comment (endv-bogdanb) [inline packages/beeq/src/components/accordion/bq-accordion.tsx:235]: We don't need `overflow-hidden` any longer ?

Comment (Cata1989) [reply packages/beeq/src/components/accordion/bq-accordion.tsx:235]: We need ... my bad ... I'll remove all of these.

**Review by dgonzalezr — DISMISSED** [review]

**Review by dgonzalezr — DISMISSED** [review]

**Review by endv-bogdanb — DISMISSED** [review]

**Pull request merged**